src/modules/Currency/Api/Guest.php

```diff
@@ -68,28 +68,35 @@ public function format($data = [])
         $convert = $data['convert'] ?? true;
         $without_currency = (bool) ($data['without_currency'] ?? false);
 
-        $p = $price;
+        $p = floatval($price);
         if ($convert) {
             $p = $price * $c['conversion_rate'];
         }
 
-        $p ??= 0;
+        if ($without_currency) {
+            return $this->select_format($p, $c['price_format']);;
+        }
+
+        // Price is negative, so we place a negative symbol at the start of the format
+        if ($p < 0) {
+            $c['format'] = '-' . $c['format'];
+        }
+
+        // Get the absolute value of the price so it displays normally for both positive and negative prices and then properly format it
+        $p = abs($p);
+        $p = $this->select_format($p, $c['price_format']);
```

Comment:
Based on the removal of the unneeded assignment previously, I was wondering, would this syntax be too much?
```suggestion
        $p = $this->select_format(abs($p), $c['price_format']);
```

Reply:
Either is fine